mac_pw_pool/nightly_maintenance.sh

```diff
@@ -22,9 +22,21 @@ if [[ ! -r "$CRONLOG" ]] || [[ ! -r "$CRONSCRIPT" ]] || [[ ! -d "../.git" ]]; th
 fi
 
 relaunch_web_container() {
-  # Assume code has changed, restart container w/ latest image
+  # Assume code change or image update, restart container.
   (
+    # Prevent podman and/or sub-processes from inheriting the lock FD.
+    # This would deadlock all future runs of this script or Cron.sh
+    # Can't use `flock --close ...` here because it "hangs" in this context.
+    for fd_path in /proc/self/fd/*; do
+        fd_nr=$(basename $fd_path)
+        [[ $fd_nr -ge 3 ]] || \
+            continue
```

Comment:
Why this usage? Why not a simple `if`? (Just curious. My brain does not process `test ||` as easily as it does `if/then`).

Reply:
agreed I also find if else easier to read in scripts

Reply:
My brain has an easier time understanding this as a "filter", "only greater/equal to 3 may proceed".   If it were a more complex condition in a different context, I'd probably agree with you though :smile: